src/components/vm/vmActions.jsx

```diff
@@ -56,6 +59,124 @@ import store from "../../store.js";
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+const onForceReboot = (vm) => domainForceReboot({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to force reboot"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+const onShutdown = (vm, setOperationInProgress) => domainShutdown({ name: vm.name, id: vm.id, connectionName: vm.connectionName })
+        .then(() => !vm.persistent && cockpit.location.go(["vms"]))
+        .catch(ex => {
+            setOperationInProgress(false);
+            store.dispatch(
+                updateVm({
+                    connectionName: vm.connectionName,
+                    name: vm.name,
+                    error: {
+                        text: cockpit.format(_("VM $0 failed to shutdown"), vm.name),
+                        detail: ex.message,
+                    }
+                })
+            );
+        });
+
+const onPause = (vm) => domainPause({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to pause"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+const onResume = (vm) => domainResume({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to resume"), vm.name),
+                detail: ex.message,
+            }
+        })
+    );
+});
+
+const onForceoff = (vm) => domainForceOff({ name: vm.name, id: vm.id, connectionName: vm.connectionName })
+        .then(() => !vm.persistent && cockpit.location.go(["vms"]))
+        .catch(ex => {
+            store.dispatch(
+                updateVm({
+                    connectionName: vm.connectionName,
+                    name: vm.name,
+                    error: {
+                        text: cockpit.format(_("VM $0 failed to force shutdown"), vm.name),
+                        detail: ex.message,
+                    }
+                })
+            );
+        });
+
+const onSendNMI = (vm) => domainSendNMI({ name: vm.name, id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+    store.dispatch(
+        updateVm({
+            connectionName: vm.connectionName,
+            name: vm.name,
+            error: {
+                text: cockpit.format(_("VM $0 failed to send NMI"), vm.name),
+                detail: ex.message,
```

Comment:
These 8 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1069-20230622-105949-1466199c-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)